src/main/java/fxlauncher/ParametersWithAdditionalNamedParameters.java

```diff
@@ -0,0 +1,41 @@
+package fxlauncher;
+
+import javafx.application.Application;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ParametersWithAdditionalNamedParameters extends Application.Parameters {
+
+    private Application.Parameters delegate;
+    private Map<String, String> combinedNamedParameters;
+
+    public ParametersWithAdditionalNamedParameters(Application.Parameters delegate, Map<String, String> additionalNamedParameters) {
+        this.delegate = delegate;
+        this.combinedNamedParameters = combineNamedParameters(delegate, additionalNamedParameters);
+    }
+
+    private Map<String, String> combineNamedParameters(Application.Parameters delegate, Map<String, String> additionalNamedParameters) {
+        Map<String, String> combinedNamedParameters = new HashMap<>();
+        combinedNamedParameters.putAll(delegate.getNamed());
+        combinedNamedParameters.putAll(additionalNamedParameters);
```

Comment:
is it OK that one can override existing parameters with additionalNamedParameters?

Reply:
We even need this to be able to inject uri after starting the application

Reply:
Good